agent/protocol/rocketmq/rocketmq.go

```diff
@@ -0,0 +1,140 @@
+package rocketmq
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"kyanos/agent/buffer"
+	"kyanos/agent/protocol"
+	"kyanos/bpf"
+)
+
+func init() {
+	protocol.ParsersMap[bpf.AgentTrafficProtocolTKProtocolRocketMQ] = func() protocol.ProtocolStreamParser {
+		return &RocketMQStreamParser{}
+	}
+}
+
+func (r *RocketMQMessage) FormatToString() string {
+	return fmt.Sprintf("base=[%s] command=[%s] payload=[%s]", r.FrameBase.String(), "todo", r.Body)
+}
+
+func (r *RocketMQMessage) IsReq() bool {
+	return r.isReq
+}
+
+func (r *RocketMQStreamParser) ParseStream(streamBuffer *buffer.StreamBuffer, messageType protocol.MessageType) protocol.ParseResult {
+	buffer := streamBuffer.Head().Buffer()
+	if len(buffer) < 8 {
+		return protocol.ParseResult{
+			ParseState: protocol.NeedsMoreData,
+		}
+	}
+
+	frameSize := int(binary.BigEndian.Uint32(buffer[:4]))
+	if frameSize > len(buffer) {
+		return protocol.ParseResult{ParseState: protocol.NeedsMoreData}
+	}
+
+	headerLength := binary.BigEndian.Uint32(buffer[4:8])
+	headerDataLen := headerLength & 0xFFFFFF
+	serializedType := byte((headerLength >> 24) & 0xFF)
+
+	if len(buffer) < 8+int(headerDataLen) {
+		return protocol.ParseResult{ParseState: protocol.NeedsMoreData}
+	}
+
+	headerBody := buffer[8 : 8+headerDataLen]
+	body := buffer[8+headerDataLen : frameSize]
+	message, err := r.parseHeader(headerBody, serializedType)
+	if err != nil {
+		return protocol.ParseResult{ParseState: protocol.Invalid, ReadBytes: int(frameSize)}
+	}
+
+	message.Body = body
+	message.isReq = messageType == protocol.Request
+	fb, ok := protocol.CreateFrameBase(streamBuffer, frameSize)
+
+	if !ok {
+		return protocol.ParseResult{
+			ParseState: protocol.Ignore,
+			ReadBytes:  frameSize,
+		}
+	} else {
+		message.FrameBase = fb
+		return protocol.ParseResult{
+			ParseState:     protocol.Success,
+			ReadBytes:      frameSize,
+			ParsedMessages: []protocol.ParsedMessage{message},
+		}
+	}
```

Comment:
It can be simplified

```suggestion
	if !ok {
		return protocol.ParseResult{
			ParseState: protocol.Ignore,
			ReadBytes:  frameSize,
		}
	}

	message.FrameBase = fb
	return protocol.ParseResult{
		ParseState:     protocol.Success,
		ReadBytes:      frameSize,
		ParsedMessages: []protocol.ParsedMessage{message},
	}
```